docs/Versioning.md

```diff
@@ -53,13 +66,33 @@ As a general principle, implementations SHOULD be robust in the face of missing
 
 By way of example, the `ModelProto.ir_version` MUST be present in every model.  The ONNX checker (`onnx/checker.py`) will enforce these rules.
 
-ISSUE: decide and document how we want to handle changes to the format that are forward compatible but will cause the generated code to introduce a breaking change (e.g., changing the data type of a field from int64 to int32, int64 to double or bytes to string).  I would recommend that we never do it after we hit 1.0.0 - prior to that we should be fairly liberal with changes to size (e.g., `int32<>int16`, `float<>double`) and conservative with changes to value space (e.g., `integral<>floatingpoint`, `string<>int`, `scalar<>message`)
+Because onnx.proto is expected to be consumed by multiple independent developers, changes to onnx.oroto SHOULD NOT break code that depends on generated language bindings (e.g., changing the type of an existing field).
 
 ISSUE: define type compatiblity rules either here or under model versioning - probably here
 
 ## Operator versioning
 
-ISSUE: sort out the design and then document.
+ONNX is defined such that the IR can evolve independently from the set of operators. In ONNX, operators represent both the signature and semantics of a given operation.  Operators are abstract interfaces in that they do not imply a specific implementation; rather, they are simply the contract between a model author and the implementations that model may execute on. 
+
+A given operator is identified by a three-tuple: (domain, op_type, op_version), written domain.op_type:op_version in prose (e.g., com.acme.FastConv:3).  Nodes in graphs always refer
+to operators by their three-part identifier.
+
+Once an operator is published, all implementations of that operator's (domain, op_type, op_version) MUST adhere to the signature and semantics of the operator at the time of publication. 
+Any change of semantics implies a new operator, which MAY share a domain and op_type with another operator. This includes adding new behavior triggered only by previously unspecified inputs or attributes - these changes in semantics also MUST have a distinct operator id.
+
+ONNX uses operator sets to group together immutable operator specifications. An ONNX operator set
+specifies both the domain of all operators it includes, as well as an opset version. The opset version is largely independent from the version field of the operators it includes. When the enventory of a given operator set changes either by addition or removal, its opset version MUST increase. Moreover,
```

Comment:
s/enventory/inventory/

And notice that with the "since_opset_version" change, one doesn't have to call out that the opset version is "independent" of the op versions, because, well, obviously some ops are going to carry over from older ones.

Reply:
See above